server/index.js

```diff
@@ -26,6 +29,58 @@ if (process.env.NODE_ENV === 'development') {
   app.use(express.static(publicPath));
 }
 
+app.post('/api/auth/sign-up', async (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+  try {
+    const hashedPassword = await argon2.hash(password);
+    const sql = `
+      insert into "users" ("username", "hashedPassword")
+      values ($1, $2)
+      returning "userId", "username", "createdAt"
+      `;
+    const params = [username, hashedPassword];
+    const result = await db.query(sql, params);
+    const [user] = result.rows;
+    res.status(201).json(user);
+  } catch (err) {
+    console.error(err);
+  }
```

Comment:
on the server side you _always_ need to ensure that a response will be sent to the client, even if there's an error. We do this by passing caught errors to `next()`